packages/discord.js/src/structures/MessagePayload.js

```diff
@@ -131,7 +132,13 @@ class MessagePayload {
       }
     }
 
-    const components = this.options.components?.map(c => createComponent(c).toJSON());
+    const components = this.options.components?.map(c => {
+      if ('toJSON' in c) {
+        c.toJSON();
+      }
+
+      return decamelize(c);
+    });
```

Comment:
Can't we do this?
```suggestion
    const components = this.options.components?.map(c => 'toJSON' in c ? c.toJSON() : decamelize(c));
```

I'm also sure that the `c.toJSON()` call without return or assignment wasn't intentional.